app/enums/link_type.ts

```diff
@@ -48,18 +108,73 @@ const linkPatterns: [string, RegExp | undefined, LinkType][] = [
   ["pwr.edu.pl", undefined, LinkType.Default],
 ];
 
+/**
+ * Spread this in a model with 'link' and 'linkType' properties for automatic link detection configuration.
+ * Remember to add new properties instead of overriding existing ones.
+ */
+export const linkTypeAutodetectSetUp = {
+  additionalProperties: {
+    linkType: {
+      availableValues: linkTypeEnumsValues,
+      isVisible: {
+        list: true,
+        show: true,
+        filter: true,
+        edit: false,
+      },
+    },
+  },
+  additionalActions: {
+    new: {
+      before: autoReplaceLinkType,
+    },
+    edit: {
+      before: autoReplaceLinkType,
+    },
+  },
+};
+
+export interface DetectionResult {
+  type: LinkType;
+  warning?: string;
+}
+
+/**
+ *  Function to add a hook on an Admin Panel field with a link.
+ *  It will automatically set the 'linkType' property of a model with a link field.
+ *  This hook will override any user-chosen link type value and, as such,
+ *  it is advised to set 'type' property visibility to false on new and edit views.
+ *  Use the 'linkTypeAutodetectSetUp' instead of directly using this function if possible.
+ */
+export async function autoReplaceLinkType(
+  request: ActionRequest,
+): Promise<ActionRequest> {
+  if (request.payload !== undefined) {
+    if (request.payload.link !== undefined) {
+      if (typeof request.payload.link !== "string") {
+        throw new Error("Link must be a string");
+      }
+      const detectedType: DetectionResult = detectLinkType(
+        request.payload.link,
+      );
+      if (detectedType.warning !== undefined) {
+        logger.info(detectedType.warning);
```

Comment:
including some additional context (such as the fact that this was triggered by an admin panel action) would likely be wise